src/main/scala/fi/oph/koski/schema/Ammatillinen.scala

```diff
@@ -543,7 +572,9 @@ case class AmmatillisenTutkinnonOsanKorkeakouluopintoSuoritus(
   override val osasuoritukset: Option[List[KorkeakouluopintojenSuoritus]] = None,
   tyyppi: Koodistokoodiviite = Koodistokoodiviite("ammatillisentutkinnonosa", koodistoUri = "suorituksentyyppi")
 ) extends KorkeakouluopintoSuoritus
-  with AmmatillisenTutkinnonOsanSuoritus
+  with AmmatillisenTutkinnonOsanSuoritus {
+  override def withLisätiedot(lisätiedot: Option[List[AmmatillisenTutkinnonOsanLisätieto]]): AmmatillisenTutkinnonOsanKorkeakouluopintoSuoritus = this
```

Comment:
Näissä withLisätiedot-metodeissa, missä palautetaan vain this, voisi olla `throw new InternalError` tms., jos parametrina annetut uudet lisätiedot on jotain muuta kuin None tai tyhjä lista.

Näin jäisi kiinni, jos esim. joskus näihin lisätään lisätiedot, ja unohdetaan päivittää withLisätiedot-metodi, ja mahdolliset muutkin ohjelmointivirheet.